### PR TITLE
task(connector-exception): Add a possibility to pass custom variables to the ConnectorException

### PR DESCRIPTION
**Pull request by johnBgood**
## Description

Add a possibility to pass custom variables when throwing a `ConnectorException`.

This is possible using the `ConnectorExceptionBuilder`.
This comes in the form of a `Map<String,Object>`.

## Related issues

closes https://github.com/camunda/team-connectors/issues/676

### CODE REVIEW

File: connector-sdk/core/src/main/java/io/camunda/connector/api/error/ConnectorExceptionBuilder.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package io.camunda.connector.api.error;
+
+import java.util.Map;
+
+public class ConnectorExceptionBuilder {
+
+  protected String errorCode;
+  protected String message;
+  protected Throwable cause;
+  protected Map<String, Object> errorVariables;
```
Comment (sbuettner) [inline connector-sdk/core/src/main/java/io/camunda/connector/api/error/ConnectorExceptionBuilder.java:27]: They can be private right?

File: connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandlerTest.java [hunk 2]
```diff
@@ -806,6 +807,33 @@ void shouldCreateBpmnError_UsingExceptionCodeAndRawContext() {
       assertThat(result.getErrorMessage()).isEqualTo("Message: exception message");
     }
 
+    @Test
+    void shouldCreateBpmnError_UsingExceptionCodeAndErrorVariables() {
+      // given
+      var errorExpression =
+          "if error.code != null then "
+              + "{ \"errorType\": \"bpmnError\", \"code\": error.code, \"message\": \"Message: \" + error.message, \"variables\": error.errorVariables} "
+              + "else {}";
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandlerTest.java:816]: Could you also add a test that verifies that the error variables are successfully passed to the command in https://github.com/camunda/connectors/blob/e277b0f002987e992d16a1384a9e167e381cbe10/connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandler.java#L289 when there is no error expression?

File: connector-sdk/core/src/main/java/io/camunda/connector/api/error/ConnectorExceptionBuilder.java (hunk 1 above)
Comment (johnBgood) [reply connector-sdk/core/src/main/java/io/camunda/connector/api/error/ConnectorExceptionBuilder.java:27]: fixed

File: connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandlerTest.java (hunk 2 above)
Comment (johnBgood) [reply connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandlerTest.java:816]: fixed

**Review by sbuettner — APPROVED** [review]

**Pull request merged**